aggregate.R

```diff
@@ -19,40 +21,82 @@ suppressWarnings(suppressMessages(library(stringr)))
 
 opts <- docopt(doc)
 
-db <- src_sqlite(path = opts[["sqlite_file"]])
+sql_file <- opts[["sqlite_file"]]
+
+output_file <- opts[["output"]]
+
+sc_type <- opts[["sc_type"]]
+
+db <- src_sqlite(path = sql_file)
 
 image <- tbl(src = db, "image") %>%
   select(TableNumber, ImageNumber, Metadata_Plate, Metadata_Well)
 
-aggregate_objects <- function(compartment) {
+# If sc_type is specified, group by different columns
+if (sc_type == "none") {
+  strata_cols <- c("Metadata_Plate", "Metadata_Well")
+} else {
+  strata_cols <- c("TableNumber", "ImageNumber", "ObjectNumber", "Metadata_Plate",
+                   "Metadata_Well")
+}
+
+aggregate_objects <- function(compartment,
+                              strata_cols = c("Metadata_Plate", "Metadata_Well"),
+                              sc_type = "none") {
   object <- tbl(src = db, compartment)
 
   object %<>% inner_join(image, by = c("TableNumber", "ImageNumber"))
 
+  if (sc_type == "isolated") {
+    object <- object %>% dplyr::filter(Cells_Neighbors_NumberOfNeighbors_Adjacent == 0)
```

Comment:
@gwaygenomics Does shouldn't works when `compartment` is `Cytoplasm` or `Nuclei`, correct?

Reply:
I originally wrote this (see below), but then realized I am wrong. This may have been my original plan, but I do not remember 100%

you are correct, this will fail as currently implemented! This is because we're aggregating compartment by compartment. If we merge everything first and then aggregate, it should work. It seems like some version of this must have worked at one point (because the call on L72 would have errored out for other non-cell compartments).

## Original thought that is wrong

If I'm remembering correctly, `sc_type` should be specified only when `compartment = "cells"` ([here](https://github.com/broadinstitute/cytominer_scripts/pull/30/files/3b72f6c53e7d8a95d89c8740ddddb36be8a0971b#diff-74eeb9ecf93aa8a5256641e5c7605ee0R73)).

Since we're doing an inner join on [L72](https://github.com/broadinstitute/cytominer_scripts/pull/30/files/3b72f6c53e7d8a95d89c8740ddddb36be8a0971b#diff-74eeb9ecf93aa8a5256641e5c7605ee0R72), this should take care of filtering for other